examples_apps/run_tutorial_similarity.py

```diff
@@ -170,57 +170,79 @@
+# of Strawberry Fields:
 #
 # 1. Through sampling.
-# 2. Using a Monte Carlo estimate of the probability.
+# 2. Using exact probability calculations.
+# 3. Using a Monte Carlo estimate of the probability.
 #
 # In the first method, all one needs to do is generate some GBS samples from the graph of
-# interest and fix the composition of the feature vector. For example, for a feature vector
-# :math:`f_{\mathbf{k} = (2, 4, 6), n_{\max}=2}` we use:
+# interest and fix the composition of the feature vector. For example, to obtain feature vector
+# :math:`f_{\mathbf{k} = (2, 4, 6), n_{\max}=2}` for the first MUTAG graph, we use:
 
-print(similarity.feature_vector_sampling(m0, event_photon_numbers=[2, 4, 6], max_count_per_mode=2))
+print(similarity.feature_vector_events_sampling(m0, [2, 4], 2))
 
 ##############################################################################
-# For the second method, suppose we want to calculate the event probabilities exactly rather than
-# through sampling. To do this, we consider the event probability :math:`p_{k, n_{\max}}` as the
-# sum over all sample probabilities in the event. In GBS, each sample probability is determined by
-# the hafnian of a relevant sub-adjacency matrix. While this is tough to calculate, what makes
-# calculating :math:`p_{k, n_{\max}}` really challenging is the number of samples the corresponding
-# event contains! For example, the 6-photon event over 17 modes :math:`E_{k=6, n_{\max}=2}`
+# We can also use any orbits of our choice instead of events:
+print(similarity.feature_vector_orbits_sampling(m0, [[1, 1], [2], [1, 1, 1, 1], [2, 1, 1]]))
+
+##############################################################################
+# For the second method, we calculate the orbit probabilities exactly rather than
+# through sampling. Considering a feature vector of orbit probabilities,
+# the probability of a single orbit :math:`p(O)` is given by:
+#
+# .. math::
+#     p(O) = \sum_{S \in O} p(S)
+#
+# where :math:`S` represents a GBS output click pattern. Calculating each :math:`p(S)` requires
+# computing a `hafnian <https://the-walrus.readthedocs.io/en/latest/hafnian.html>`__, which gets
+# exponentially difficult with increasing photon number. Exact probability of an event
+# :math:`p_{k,n_{\max}}` can be calculated similarly or by summing over its constituent orbit probabilities.
+#
+# Built-in functions :func:`~.feature_vector_orbits` and :func:`~.feature_vector_events`
+# can be used to get exact feature vectors. These functions
+# use a keyword argument ``samples`` to signal producing either exact or approximate probabilities.
+# ``samples`` is set to ``None`` to get exact feature vector by default. To use Monte Carlo estimation,
+# ``samples`` can be set to the number of samples desired to be used in the estimation, as shown later.
+# For example, to get the exact event probabilities in the feature vector example
+# :math:`f_{\mathbf{k} = (2, 4, 6), n_{\max}=2}` seen previously, we use:

```

Comment:
should we just keep using [2, 4, 6] like before? I changed it only for faster documentation building but it is good to be consistent; either all examples use [2,4] or [2,4,6]

Reply:
I'm cautious about switching to [2, 4, 6] for the exact calculation because it was really slow on my machine. Does it change the story much if we change everything to [2, 4]? This is a bit more work though.

Reply:
no worries. it is not too much. I can change it to [2,4] everywhere :)